Getting NPE in HandleError

Please answer these questions before submitting your issue. Thanks!

### What version of Cassandra are you using?


### What version of Gocql are you using?
Commit as of Feb 8th

### What did you do?


### What did you expect to see?


### What did you see instead?
Getting the following NPE

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x9a3c31]

 goroutine 72571 [running]:
 github.com/gocql/gocql.(*controlConn).HandleError(0xc002ce8c80, 0xc0024e9f00, 0xd20120, 0xc002e95940, 0xc002e95901)
 /go/pkg/mod/github.com/gocql/gocql@v0.0.0-20190208221138-c53c3654dc8a/control.go:393 +0x71
github.com/gocql/gocql.(*Conn).closeWithError(0xc0024e9f00, 0xd20120, 0xc002e95940)
/go/pkg/mod/github.com/gocql/gocql@v0.0.0-20190208221138-c53c3654dc8a/conn.go:465 +0x273
github.com/gocql/gocql.(*Conn).heartBeat(0xc0024e9f00)
/go/pkg/mod/github.com/gocql/gocql@v0.0.0-20190208221138-c53c3654dc8a/conn.go:520 +0x2d5
created by github.com/gocql/gocql.(*Session).dial
/go/pkg/mod/github.com/gocql/gocql@v0.0.0-20190208221138-c53c3654dc8a/conn.go:260 +0x789

This relates to line 393 in the following file (as of Feb 8th)
https://github.com/gocql/gocql/blob/33c0e89ca93ad9f5069645af476dfb1f97195621/control.go

>	if oldConn.conn != conn {

Here for some reason the oldConn seems to be nil and so the NPE.

---

If you are having connectivity related issues please share the following additional information

### Describe your Cassandra cluster
please provide the following information

- output of `nodetool status`
Datacenter: datacenter1
=======================
Status=Up/Down
|/ State=Normal/Leaving/Joining/Moving
--  Address     Load       Tokens       Owns    Host ID                               Rack
UN  172.21.0.2  178.87 MB  256          ?       df9a772e-1de9-4992-abc9-a3538369e2d0  rack1

- output of `SELECT peer, rpc_address FROM system.peers`

cqlsh> SELECT peer, rpc_address FROM system.peers;

 peer | rpc_address
------+-------------

(0 rows)

- rebuild your application with the `gocql_debug` tag and post the output

--- a/control.go
+++ b/control.go
@@ -389,7 +389,10 @@ func (c *controlConn) HandleError(conn *Conn, err error, closed bool) {
 	}
 
 	oldConn := c.getConn()
-	if oldConn.conn != conn {
+
+	// If connection has long gone, and not been attempted for awhile,
+	// it's possible to have oldConn as nil here (#1297).
+	if oldConn != nil && oldConn.conn != conn {
 		return
 	}
 

